src/Data/Common/UrlState/useUrlStateWithCurrentPage.ts

```diff
@@ -0,0 +1,50 @@
+import { isEqual, isArray } from "lodash";
+import { provide, Location, Replace, StateConfig, Update } from "./helpers";
+import { handleUrlState } from "./useUrlState";
+
+export const useUrlStateWithCurrentPage = provide(
+  handleUrlStateWithCurrentPage,
+);
+const valueIsValid = (value: unknown): value is string[] => {
+  return isArray(value) && value.every((param) => typeof param === "string");
+};
+
+const from = (value: string[]): CurrentPage => {
+  return { kind: "CurrentPage", value: valueIsValid(value) ? value : [] };
+};
+const serialize = (currentPage: CurrentPage): string[] => {
+  return currentPage.value;
+};
+
+const parse = (candidate: unknown): CurrentPage | undefined => {
+  return valueIsValid(candidate) ? from(candidate) : undefined;
+};
+
+export interface CurrentPage {
+  kind: "CurrentPage";
+  value: string[];
```

Comment:
Why did you go for the array of strings?

Reply:
There are Two sets of two arguments responsible for dictating on which page we are(depending if we are going froward or backward), separated by "&" so I thought that if they are always in pairs, then instinctively picked this one.